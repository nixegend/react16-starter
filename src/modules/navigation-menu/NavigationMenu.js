import React from 'react';
import PropTypes from 'prop-types';
import { Link, NavLink } from 'react-router-dom';

import internalUrls from '@constants/internal-urls';

const NavigationMenu = ({ className }) => (
  <nav className={className}>
    <Link to={internalUrls.HOME.path} className="logo">
      <img alt="logo" src="/images/sigma-logo.svg" />
    </Link>
    <NavLink to={internalUrls.HOME.path} exact>
      Home
    </NavLink>
    <NavLink to={internalUrls.ENTITY.path}>Entity</NavLink>
    <NavLink to={internalUrls.TEST.path}>Test</NavLink>
  </nav>
);

NavigationMenu.propTypes = {
  className: PropTypes.string
};

NavigationMenu.defaultProps = {
  className: ''
};

export default NavigationMenu;
