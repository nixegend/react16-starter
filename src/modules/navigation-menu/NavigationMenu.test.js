
import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import NavigationMenu from './NavigationMenu';

describe('<NavigationMenu />', () => {
  it('Shoud render correctly NavigationMenu component', () => {
    const tree = renderer.create(<Router><NavigationMenu /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
