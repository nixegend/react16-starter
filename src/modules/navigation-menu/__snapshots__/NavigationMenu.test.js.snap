// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationMenu /> Shoud render correctly NavigationMenu component 1`] = `
<nav
  className=""
>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="logo"
      src="/images/sigma-logo.svg"
    />
  </a>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Home
  </a>
  <a
    aria-current={null}
    href="/entity"
    onClick={[Function]}
  >
    Entity
  </a>
  <a
    aria-current={null}
    href="/test"
    onClick={[Function]}
  >
    Test
  </a>
</nav>
`;
